src/mmw/js/src/user/models.js

```diff
@@ -42,157 +45,140 @@ var UserModel = Backbone.Model.extend({
     }
 });
 
-var LoginFormModel = Backbone.Model.extend({
+var ModalBaseModel = Backbone.Model.extend({
+    defaults: {
+        success: false,
+        client_errors: null,
+        server_errors: null
+    }
+});
+
+var LoginFormModel = ModalBaseModel.extend({
     defaults: {
-        validationErrors: null,
         username: null,
-        password: null,
-        loginError: null
+        password: null
     },
 
-    validate: function(attrs, options) {
-        var errors = [];
-
-        this.set('validationErrors', null);
-        this.set('loginError', null);
+    url: '/user/login',
 
-        if (!attrs.username) {
-            errors.push('Please enter a username');
-        }
+    validate: function(attrs) {
+        var errors = [];
 
-        if (!attrs.password) {
-            errors.push('Please enter a password');
-        }
+        if (!attrs.username) errors.push('Please enter a username');
+        if (!attrs.password) errors.push('Please enter a password');
 
-        if (errors.length >= 1) {
-            this.set('validationErrors', errors);
+        if (errors.length) {
+            this.set({
+                'client_errors': errors,
+                'server_errors': null
+            });
             return errors;
+        } else {
+            this.set({
+                'client_errors': null,
+                'server_errors': null
+            })
         }
     }
 });
 
-var SignUpFormModel = Backbone.Model.extend({
+var SignUpFormModel = ModalBaseModel.extend({
     defaults: {
         username: null,
         password1: null,
         password2: null,
         email: null,
-        agreed: false,
-        clientErrors: null,
-        serverErrors: null,
-        success: false
+        agreed: false
     },
 
     url: '/user/sign_up',
 
-    validate: function(attrs, options) {
+    validate: function(attrs) {
         var errors = [];
 
-        this.set({
-            'clientErrors': null,
-            'serverErrors': null
-        });
-
-        if (!attrs.username) {
-            errors.push('Please enter a username');
-        }
-
-        if (!attrs.email) {
-            errors.push('Please enter an email addresss');
-        }
-
-        if (!attrs.password1) {
-            errors.push('Please enter a password');
-        }
-
-        if (!attrs.password2) {
-            errors.push('Please repeat the password');
-        }
-
-        if (!attrs.agreed) {
-            errors.push('Please check the agreement');
-        }
-
-        if (errors.length >= 1) {
-            this.set('clientErrors', errors);
+        if (!attrs.username) errors.push('Please enter a username');
+        if (!attrs.email) errors.push('Please enter an email address');
+        if (!attrs.password1) errors.push('Please enter a password');
+        if (!attrs.password2) errors.push('Please repeat the password');
+        if (!(attrs.password1 === attrs.password2))
+            errors.push('Passwords do not match');
```

Comment:
It might seem silly, but it's a good habit to get into to use curly braces even for conditionals with only 1 statement.